### PR TITLE
Display thumbnail previews for images in conversation list.

### PR DESCRIPTION
**Pull request by moxie0**
conversation-list-thumbnails

### CODE REVIEW

**Comment by mcginty** [comment]
Since `ThreadDatabase.update()` is called every single time a message is inserted or updated I'm worried about the extra time this will add to our latency in messages appearing in the conversation on slow-I/O devices. Have you checked how much time this adds onto the `insertMessageOutbox` operation for example?

**Comment by moxie0** [comment]
The slidedeck query has always been running as part of the thread update operation, we're just checking the results of it.  It's an indexed query that I assume should be fast, but I just benchmarked how long the blocking operation for retrieving the slide takes in the thread update.  On my Nexus 5 it was between 1-6 milliseconds, which I'm pretty sure is the cost of the synchronized block.  =)

**Comment by moxie0** [comment]
Also, this moves the thread update out of the message insertion commit, so the change should actually improve time between sent and cursor update.

File: src/org/thoughtcrime/securesms/ConversationListItem.java [hunk 1]
```diff
@@ -136,6 +145,23 @@ public int getDistributionType() {
...
+
+      LayoutParams subjectParams = (RelativeLayout.LayoutParams)this.subjectView.getLayoutParams();
+      subjectParams.addRule(RelativeLayout.LEFT_OF, R.id.thumbnail);
+      this.subjectView.setLayoutParams(subjectParams);
+    } else {
+      this.thumbnailView.setVisibility(View.GONE);
+
+      LayoutParams subjectParams = (RelativeLayout.LayoutParams)this.subjectView.getLayoutParams();
+      subjectParams.addRule(RelativeLayout.LEFT_OF, 0);
+      this.subjectView.setLayoutParams(subjectParams);
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationListItem.java]: seems like we shouldn't be using a RelativeLayout parent if this is necessary to make it work

Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationListItem.java]: i think relativelayout is appropriate, but you have to specify the leftof constraint dynamically in order to get the text snippet to terminate at the thumbnail without it terminating at the date when the thumbnail is gone.

**Pull request merged**